### PR TITLE
LPS-184220 does not expire KB Article if the status is pending

### PR DESCRIPTION
**Pull request by AliciaGarciaGarcia**
https://issues.liferay.com/browse/LPS-184220

**Steps to reproduce:**

1. Go to system settings > knowledge base
2. Update the Check Interval to 1
3. Enable single approve workflow for knowledge base
4. Go to knowledge base > Add an article with expire date (2mins later)
5. Wait 2 mins or longer
6. Refresh the page
7. Observe the added article status

**Expected Result:**
The article won't expire until is approved

- LPS-184220 does not expire KB Article if the status is pending
- LPS-184220 no needed
- LPS-184220 do not show the expiring soon label if status is pending

### CODE REVIEW

**Comment by liferay-lima** [comment]
ci:test:sf

**Comment by liferay-lima** [comment]
ci:test:relevant

**Comment by AliciaGarciaGarcia** [comment]
ci:stop

**Comment by Hanlf** [comment]
ci:test:relevant

File: modules/apps/knowledge-base/knowledge-base-web/src/main/resources/META-INF/resources/admin/view_descriptive.jsp [hunk 1]
```diff
@@ -147,7 +147,12 @@ KBArticleViewDisplayContext kbArticleViewDisplayContext = new KBArticleViewDispl
 								<span class="label-item label-item-expand"><liferay-ui:message key="expiring-soon" /></span>
 							</span>
 
-							<liferay-ui:icon-help message='<%= kbArticle.getExpirationDate()!= null ? dateFormatDateTime.format(kbArticle.getExpirationDate()) : "" %>' />
+							<clay:icon
```
Comment (adolfopa) [inline modules/apps/knowledge-base/knowledge-base-web/src/main/resources/META-INF/resources/admin/view_descriptive.jsp:150]: Can you extract the value of `aria-label` and `title` to a variable and reuse it? Same thing in `view_table.jsp`.

Comment (adolfopa) [reply modules/apps/knowledge-base/knowledge-base-web/src/main/resources/META-INF/resources/admin/view_descriptive.jsp:150]: Inlining expressions is fine as long they're "free". For example, simple getters are inlined by the JIT and transformed into direct field references; in this case, even though it is not superexpensive to inline it, it is a bit wasteful.

**Review by adolfopa — CHANGES_REQUESTED** [review]

File: modules/apps/knowledge-base/knowledge-base-web/src/main/resources/META-INF/resources/admin/view_descriptive.jsp (hunk 1 above)
Comment (AliciaGarciaGarcia) [reply modules/apps/knowledge-base/knowledge-base-web/src/main/resources/META-INF/resources/admin/view_descriptive.jsp:150]: That was my first approach, but later I remembered some times that after that an additional commit was added by brian inlining it.
But I will extract a variable, If you and I feel that is the correct way, we should do it.

Thanks!

**Comment by AliciaGarciaGarcia** [comment]
added https://github.com/liferay-lima/liferay-portal/pull/4307/commits/83e9e69a077c13821738db42574ac2601f1f3716 extracting variable

**Comment by AliciaGarciaGarcia** [comment]
ci:test:sf

**Comment by AliciaGarciaGarcia** [comment]
ci:test:relevant

**Comment by AliciaGarciaGarcia** [comment]
SF problem the tag has no replace for some attributes
asking the FI team 
> https://liferay.slack.com/archives/CNBG06JS3/p1684325740437329

**Comment by AliciaGarciaGarcia** [comment]
@adolfopa 
a forward force will be necessary on this one if test are ok. There is no solution for now of the SF rule :(

**Comment by adolfopa** [comment]
@AliciaGarciaGarcia Can we then add an SF exclusion for that particular rule? If we forward this without it, we'll break all future CI builds until fixed.

**Comment by AliciaGarciaGarcia** [comment]
ci:stop

**Comment by AliciaGarciaGarcia** [comment]
ci:test:sf

**Comment by AliciaGarciaGarcia** [comment]
@adolfopa  I will remove the on hold tag

**Comment by AliciaGarciaGarcia** [comment]
ci:reevaluate:1397401_4786

**Comment by Hanlf** [comment]
ci:test:relevant

**Pull request closed without merging**